### PR TITLE
Feature: Huggingface dataset loader with examples

### PR DESCRIPTION
**Pull request by azamiftikhar1000**
Feature: Huggingface dataset loader with examples

### CODE REVIEW

File: langchain/document_loaders/hugging_face_dataset.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        self.data_files = data_files
+        self.cache_dir = cache_dir
+        self.keep_in_memory = keep_in_memory
+        self.save_infos = save_infos
+        self.use_auth_token = use_auth_token
+        self.num_proc = num_proc
+
+    def load(self) -> List[Document]:
+        """Load documents."""
+        try:
```
Comment (vowelparrot) [inline langchain/document_loaders/hugging_face_dataset.py:55]: Could we please check that this is installed on initialization?

**Comment by azamiftikhar1000** [comment]
@vowelparrot @hwchase17 @agola11 
I am working on enhancing this current PR by adding a new features to the Huggingface dataset loader(loading stream dataset https://github.com/hwchase17/langchain/issues/2864). I would greatly appreciate it if you could take some time to review this PR and share your thoughts. If you have any suggestions, concerns, or questions, please don't hesitate to leave comments on the PR. I'll be more than happy to address them.

**Review by hwchase17 — APPROVED** [review]
this looks great to me!

**Review by hwchase17 — COMMENTED** [review]
actually one nit (which i can make)

this should be exposes in document_loaders.__init__

**Pull request merged**